docker/README.md

```diff
@@ -79,6 +79,16 @@ as services in detached mode. The last for the client will run an
 interactive shell in the client container. See below for information
 on how to use the client container.
 
+### Build for SGX ###
+
+For the contract enclave to run in SGX hardware mode, the `pdo_services`
+image must be built using the following target:
+```bash
+    make build_sgx_services
+```
+Inside the `pdo_services` images, the `SGX_MODE` environment variable 
+can help distinguish the build type.
```

Comment:
As i think also Mic has suggested i think we should "type" the images names, not doing so is asking for trouble; _in particular_ given that contrary to the previous approach of relying on external variable definitino of SGX_MODE (which i preferred) we have now multiple targets which "conflict" as far as sgx-mode is concerned which increases opportunities for mix-ups ...

Reply:
There was previous discussion here https://github.com/hyperledger-labs/private-data-objects/pull/477/files#r1527009436.
My earlier attempt unfortunately was not promising because we have SGX_MODE in multiple images and the images are interdependent. Hence, it's difficult to do the typing without modifying makefile and yaml and docker files.

However, it is possible that SGX_MODE is only necessary in the pdo-services image. So it could be removed from the other images, and and then we simply add a new type for pdo-services in hw mode.

Conclusion:
- the BKM here is to always start with a fresh clean development environment
- mix-n-match can be a concern
- solving/mitigating this requires further work -- IMO this is orthogonal to this PR, and I'm ok to do it separately.

Reply:
> However, it is possible that SGX_MODE is only necessary in the pdo-services image. 

Previously that was the case.  However, somebody mentioned that the sdk (or, more likely, sgxssl) would be different based on mode (which on further reflection seems though rather surprising and strange).

In any case, base service is mode-agnostic, then we could (should) just remove the ENV in the corresponding dockerfile so the status is not persisted

> So it could be removed from the other images, and and then we simply add a new type for pdo-services in hw mode.

That was my thinking: we just type pdo_services with the mode.  Initially i thought that might also apply to ccf if we would run that in HW mode but according to Mic CCF always builds both version and so only the runtime SGX_MODE variable matters.

Reply:
> Why is that orthogonal? Given that you have already sgx_build and sgx_run targets, seems also easy to type that image?

Because I was working on it [here](https://github.com/bvavala/private-data-objects/tree/bruno.240315.specialize-images) and it was too complicated to solve it in this PR.
Also, I have not checked the simplifications that we have discussed above -- if those work, the modifications are way easier.

Reply:
I gave it a try and the modifications were fairly easy -- I have added a commit to type the sgx image.